File: java/src/test/java/io/binac/leetcode/JumpGameIITest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class JumpGameIITest {
    private static final JumpGameII.Solution1 SOLUTION1 = new JumpGameII.Solution1();

    private final int[] NUMS1 = {2, 3, 1, 1, 4};
    private final int EXPECTED1 = 2;

    private final int[] NUMS2 = {2, 3, 0, 1, 4};
    private final int EXPECTED2 = 2;

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.jump(NUMS1));
        assertEquals(EXPECTED2, SOLUTION1.jump(NUMS2));
    }
}

File: cpp/tests/JumpGameIITest.cpp
#include "JumpGameII.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(JumpGameII, Solution45_1) {
  auto S45_1 = Solution45_1();

  std::vector<int> Nums1{2, 3, 1, 1, 4};
  int const Expected1 = 2;
  EXPECT_EQ(Expected1, S45_1.jump(Nums1));

  std::vector<int> Nums2{2, 3, 0, 1, 4};
  int const Expected2 = 2;
  EXPECT_EQ(Expected2, S45_1.jump(Nums2));
}
